cpp/ps0004/ps0004_soin_cpp.cpp

```diff
@@ -2,17 +2,27 @@
 #include <conio.h>
 using namespace std;
 int main() {
-	int num;
+	// prompting user for number 1/1.5
+	float num;
 	cout << "Enter a number (ASCII value between 32 and 126) \n";
 	cin >> num;
 
-	if (num >= 32 && num <= 126) {
+	// checking invalidaty of user's input 1.5/4
+	// printing the number and the opposite printable character 1.5/2
+	if(!num) {
+		cout << "Invalid character/s or zero value!\n";
+	}
+	else if((int) num != num) {
+		cout << "Invalid: a floating point number!\n";
+	}
+	else if(num >= 32 && num <= 126) {
 		char ch = static_cast<char>(num);
-		cout << "Equivalent character \n" << ch << endl;
+		cout << "Equivalent character for the number " << num << " is " << ch << "." << endl;
 	}
 	else {
-		cout << "Invalid input. ASCII value must be between 32 and 126\n" << std::endl;
+		cout << "Invalid: number is out of the range!\n";
 	}
 
 	_getch();
-}
\ No newline at end of file
+}
+// good work 👍 5/10

```

Comment:
- [x] program is in the correct path
- [ ] no syntax errors, bugs or warnings while compilation and execution
- [x] formatted code to be clear and readable
- [ ] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case
you can check the [problem solution](https://onlinegdb.com/9x6wLTlkK) for more